allocator/allocator.go

```diff
@@ -318,37 +330,53 @@ func (b *batchedTxn) Checkpoint() error {
 func (b *batchedTxn) Commit() (*clientv3.TxnResponse, error) {
 	if len(b.nextCmps) != 0 || len(b.nextOps) != 0 {
 		panic("must call Checkpoint before Commit")
+	} else if len(b.ops) == 0 {
+		return nil, nil // No-op.
 	}
 
-	if r, err := b.txnDo(clientv3.OpTxn(b.cmps, b.ops, nil)); err != nil {
+	var response, err = b.txnDo(clientv3.OpTxn(b.cmps, b.ops, nil))
+
+	if log.GetLevel() >= log.DebugLevel {
+		b.debugLogTxn(response, err)
+	}
+
+	if err != nil {
 		return nil, err
-	} else if !r.Succeeded {
-		return r, fmt.Errorf("transaction checks did not succeed")
+	} else if !response.Succeeded {
+		return response, fmt.Errorf("transaction checks did not succeed")
 	} else {
 		if len(b.ops) > 0 {
```

Comment:
It seems like this check may no longer be necessary? Since we check the length of ops above and exit early it looks like this condition will always be true (unless it is possible for `txnDo` to modify `ops` to be empty, though is this still considered a NOOP?).  If this is the case we could just set `b.noop = false` near the top after the set of checks.

Reply:
Yep, you're right, this is vestigial now.